linker/Linker/Driver.cs

```diff
@@ -72,7 +74,71 @@ public static int Execute (string[] args, ILogger customLogger = null)
 
 		public Driver (string [] args)
 		{
-			_queue = new Queue<string> (args);
+			_queue = ProcessResponseFile (args);
+		}
+
+		Queue<String> ProcessResponseFile (string [] args)
+		{
+			var result = new Queue<string> ();
+			foreach (string arg in args) {
+				if (arg.StartsWith ("@")) {
+					try
+					{
+						string responseFileName = arg.Substring(1);
+						IEnumerable<string> responseFileLines = File.ReadAllLines(responseFileName).Select(line => line.Trim());
+						ParseResponseFileLines(responseFileLines, result);
+					} catch (Exception) {
+						Usage ("Cannot read response file");
+					}
+				} else {
+					result.Enqueue (arg);
+				}
+			}
+			return result;
+		}
+
+		public static void ParseResponseFileLines (IEnumerable<string> responseFileLines, Queue<string> result)
+		{
+			string responseFileText = string.Join (" ", responseFileLines);
+			int idx = 0;
+			while  (idx < responseFileText.Length) {
+				while  (idx < responseFileText.Length && char.IsWhiteSpace (responseFileText [idx])) {
+					idx++;
+				}
+				if  (idx == responseFileText.Length) {
+					break;
+				}
+				StringBuilder argBuilder = new StringBuilder ();
+				bool inquote = false;
+				while  (true) {
+					bool copyChar = true;
+					int numBackslash = 0;
+					while  (idx < responseFileText.Length && responseFileText [idx] == '\\') {
```

Comment:
Would be nice to keep the formatting consistent, you are using mix of 0, 1 or 2 spaces

Reply:
Store `responseFileText [idx]` to local variable and use it instead

Reply:
My `algorithm` for producing the formatting is the following:
Format the code in Visual Studio.
Replace all instances of `" ("` to `"("`. repeat until I can't, so there is no space before any "("
Replace all instances of `"("` to `" ("`, so that each "(" is prefixed with one space.
Do the same with "["

Since the operations are done manually, it is easy to make mistake. It would be nice if there is a formatter or something such that I can use to produce the required formatting.

Reply:
I think it will work in the scenario you are talking about - here is why:

1. Suppose the command line arguments are `@foo.rsp -c skip` and the content of `foo.rsp` is simply `-c copy`

2. The `ProcessResponseFile()` will get a `string[] args = { "@foo.rsp", "-c", "skip" }`

3. The `ProcessResponseFile` loop will get `@foo.rsp`, so it calls `ParseResponseFileLines(new string[]{"-c copy"})`

4. After the call, the queue will be `"-c","copy"`

5. Next, the `ProcessResponseFile` loop will get `"-c"` and then `"skip"`

6. When the loop is done, the final queue would be `{ "-c","copy", "-c", "skip" }`, it would look as if the overridding is not happening, but read on ...

7. The program would then proceed as if someone type in the command line `illink -c copy -c skip`.

8. When it comes to actually executing, the `Driver.run()` will start.

9. The `while (HaveMoreTokens())` loop will start execute, it process the queue in a first-in-first-out manner, which means it will see `'-c'` twice.

10. When the `"-c"` is processed for the second time, the `'c'` switch case would kick in and override the `context.CoreAction` to `skip`.

Because of point 6 - arguably - any overridding logic that need to happen with the presence with the response file will also need to happen in the plain command line case.

I do spot one thing though, suppose we specified `-c AddBypassNGen`, then the `ParseAssemblyAction` would set `_needAddBypassNGenStep`, but if we override it later, it will never set back to `false`, we probably should fix that, independent of the support for response file.

Reply:
That's was just an example. It needs to work with any order of response files and individual arguments (-c skip @foo.rsp, @foo.rsp -c skip @bar.rsp, etc.) which won't because you process response arguments always first.

Reply:
I think the arguments will eventually be processed in the specified order.

Suppose the command line is `A @b.rsp C @d.rsp E` and the content of `b.rsp` and `d.rsp` are simply `B` and `D`, the queue after the call `ProcessResponseFile()` should be `A,B,C,D,E`. In particular, the enqueue operation on line 94 should make sure `A,C,E` are all populated at the right spot.

After the `ProcessResponseFile()` call, the arguments will be processed in the `A,B,C,D,E` order, which I believe is what we wanted.

Reply:
> Store `responseFileText [idx]` to local variable and use it instead

It appears to me that `responseFileText [idx]` is frequently used in loop conditions and `idx` got changed all the time. I am thinking about not fixing this comment for now. 

If you feel like we must fix it, let me know why and I will work on it in the next iteration.